pages/your_tasks/readable_code.md

```diff
@@ -0,0 +1,99 @@
+what I have done and why?" 
+Or: "If a new person who has just joined the project reads my code, will they be able to understand what I have written?"
+
+In addition, software requirements will likely change over time because software is dynamic in nature. 
+When these requirements change, a developer (who is not necessarily the same person who wrote the original code) needs to make changes to the software
+to satisfy these new and emerging needs. 
+They do this by reading the original code to understand it and identify what needs to change. 
+Readable code facilitates the the process of understanding the code, the evolution of the codebase, and saves future developers' time and effort.
+
+### Considerations
+
+- Readable code is easier to understand, maintain, debug and extend (reuse) - saving time and effort.
+- Readable code can also make it better and more secure in a way, as it is easier for reviewers to pick out errors.
+
+
+## What practices are there to help me improve the readability of my code?
+
+One of the most important things we can do to make sure our code is readable by others (and ourselves a few months down the line) is to make sure that it is descriptive, 
+cleanly and consistently formatted and uses sensible, descriptive names for variable, function and module names. 
+In order to help us format our code, we generally follow guidelines known as a style guide. 
+A style guide is a set of conventions that we agree upon with our colleagues or community, to ensure that everyone contributing to the same project is producing code which 
+looks similar in style. 
+While a group of developers may choose to write and agree upon a new style guide unique to each project, in practice many programming languages have a single style guide 
+which is adopted almost universally by the communities around the world. 
+
+Structure code that is modular and split into small, reusable functions.
+Use the common code pattern for creating software that can read input from command line
+Follow best practices in structuring code and organising software/research project directories for improved readability, accessibility and reproducibility.
+
+
+### Considerations
+
+- Code is read much more often than it is written - style guidelines are intended to improve the readability of code and make it consistent across the wide spectrum of code
+- Consistency with the style guide is important. Consistency within a project is more important. Consistency within one module or function is the most important. However, know when to be inconsistent - sometimes style guide recommendations are just not applicable. When in doubt, use your best judgment. Look at other examples and decide what looks best.
+
+
+### Solutions
+
+Here are some of the practices to follow to make your code more readable:
+
+- Use consistent formatting - consistent indentation and spacing make code easier to understand and parse. 
```

Comment:
https://black.readthedocs.io/en/stable/ is an example - python again of a code formatter than can help - is it worth putting examples in?

Reply:
Modern IDEs have such support built in - I'll mention this

Reply:
Done via 172c1d84866e94df939e5846276d2a5ab88ccb13.